spec/features/import_tpas_satisfaction_data_spec.rb

```diff
@@ -0,0 +1,54 @@
+require 'rails_helper'
+require 'importers'
+require 'mail_retriever'
+
+RSpec.feature 'Importing TPAS data' do
+  scenario 'Storing satisfaction data' do
+    when_i_import_tpas_data
+    then_the_satisfaction_data_has_been_saved
+  end
+
+  scenario 'Re-importing satisfaction data does not create duplicates' do
+    when_i_import_tpas_data
+    and_i_import_tpas_data_a_second_time
+    then_the_satisfaction_data_has_been_saved
+  end
+
+  def when_i_import_tpas_data
+    setup_imap_server(
+      attachment:  File.read(Rails.root.join('spec/fixtures/tpas_satisfaction.csv'), mode: 'rb')
+    )
+    Importers::TpasSatisfaction::Importer.new.import
+  end
+
+  def and_i_import_tpas_data_a_second_time
+    Importers::TpasSatisfaction::Importer.new.import
+  end
+
+  def setup_imap_server(attachment:)
+    mail_attachment = double( # Replace with local IMap server
+      :mail_attachment,
+      file: StringIO.new(attachment),
+      uid: SecureRandom.uuid
+    )
+    allow_any_instance_of(MailRetriever).to receive(:search).and_return([mail_attachment])
```

Comment:
I cringe at `allow_any_instance_of`. It's presence hints at a smell, usually an opaque dependency. Can we inject a double for this somewhere instead?